Remove dependency on RemoteExecutor on some files

ricardobossan opened the pull request:
<!-- Please read CONTRIBUTING.md before submitting a pull request -->

Fixes #


## Proposed changes

- 
- 
- 

<!-- We are in TELL-MODE the following section must be completed -->

## Customer Impact

- 
- 

## Regression? 

- Yes / No

## Risk

-

<!-- end TELL-MODE -->


## Screenshots <!-- Remove this section if PR does not change UI -->

### Before

<!-- TODO -->

### After

<!-- TODO -->


## Test methodology <!-- How did you ensure quality? -->

- 
- 
- 

## Accessibility testing  <!-- Remove this section if PR does not change UI -->

<!--
     Microsoft prioritizes making our products accessible. 
     WinForms has a key role in allowing developers to create accessible apps. 
     
     When submitting a change which impacts UI in any way, including adding new UI or
     modifying existing controls the developer needs to run the Accessibility Insights
     tool (https://accessibilityinsights.io/) and verify that there are no changes or
     regressions. 
     
     The developer should run the Fast Pass over the impacted control(s) and provide
     a snapshot of the passing results along with before/after snapshots of the UI.
     More info: (https://accessibilityinsights.io/docs/en/web/getstarted/fastpass)
  -->


 

## Test environment(s) <!-- Remove any that don't apply -->

- <!-- use `dotnet --info` -->


<!-- Mention language, UI scaling, or anything else that might be relevant -->

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12379)

Tanya-Solyanik commented on src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Application.ParkingWindowTests.cs:
```diff
@@ -1,17 +1,16 @@
-        using RemoteInvokeHandle invokerHandle = RemoteExecutor.Invoke(() =>
+        Task.Run(() =>
         {
             Control.CheckForIllegalCrossThreadCalls = true;
```
Because this test modifies a static value, it should run in a sequential collection( move it to a new file, like Leaf did) and it should set the static back to the initial value using a try/fianally blocks

ricardobossan commented:
/azp run

Tanya-Solyanik commented on src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Application.ParkingWindowTests.cs:
```diff
@@ -28,10 +27,9 @@ public void ParkingWindow_DoesNotThrowOnGarbageCollecting()
 
-        // verify the remote process succeeded
-        Assert.Equal(RemoteExecutor.SuccessExitCode, invokerHandle.ExitCode);
+        Control.CheckForIllegalCrossThreadCalls = false;
```
This should be in a try/finally block

ricardobossan commented:
@Tanya-Solyanik  I have re-enabled all tests that were previously skipped with the explanation:  
`Crash with AbandonedMutexException. See: https://github.com/dotnet/arcade/issues/5325`.  

Additionally, I restored the correct state of the `Application.ParkingWindowTests.cs` method by setting:  
`Control.CheckForIllegalCrossThreadCalls = true;`.

### Summary of Attempts:

- **Approach 1:**  
  Removed `RemoteExecutor` only from `Application.ParkingWindowTests.cs` while simply re-enabling the other tests.  
  - [CI Checks Results](https://github.com/user-attachments/files/18466297/Remove_Parking_only.txt)

- **Approach 2:**  
  Removed `RemoteExecutor` from all tests.  
  - [CI Checks Results](https://github.com/user-attachments/files/18466296/Remove_All.txt)

Both approaches passed local tests but failed automated CI checks.  

For now, I’m pausing work on this issue to focus on higher-priority tasks. Let me know if you have any suggestions or would like me to revisit this sooner!

Tanya-Solyanik commented:
We'll re-open when we are working on this again.
